app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java

```diff
@@ -224,41 +211,41 @@ else if(ec4>=2)
             @Override
             public void onClick(View v) {
                 ecmsg++;
-                if(ecmsg==1)
-                {
+                if (ecmsg == 1) {
                     tmsg.setImageDrawable(ContextCompat.getDrawable(getContext(), R.drawable.dark));
                     emsg.setEnabled(true);
-                }
-                else if(ecmsg>=2)
-                {
+                } else if (ecmsg >= 2) {
                     String msg = emsg.getText().toString();
                     editormsg.putString("nummsg", msg);
                     editormsg.apply();
 
-                    String valuemsg = nummsg.getString("nummsg","Enter your message");
+                    String valuemsg = nummsg.getString("nummsg", "Enter your message");
                     emsg.setText(valuemsg);
 
                     tmsg.setImageDrawable(ContextCompat.getDrawable(getContext(), R.drawable.light));
                     emsg.setEnabled(false);
-                    ecmsg=0;
+                    ecmsg = 0;
                 }
 
 
             }
         });
 
 
-
         send.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
 
-                String number1 = num1.getString("num1","Enter Num1");
-                String number2 = num2.getString("num2","Enter Num2");
-                String number3 = num3.getString("num3","Enter Num3");
-                String number4 = num4.getString("num4","Enter Num4");
+                // A single toast efficiently conveys the message to the user.
+                Toast.makeText(getActivity(), "Message Sent",
+                        Toast.LENGTH_LONG).show();
+
+                String number1 = num1.getString("num1", "Enter Num1");
```

Comment:
We can refactor these strings vars into a list of strings for better scalability in future.